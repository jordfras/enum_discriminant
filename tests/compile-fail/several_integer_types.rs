use enum_discriminant::discriminant;

#[discriminant(u8, u32)]
enum SeveralIntegerTypes {
    Variant,
}

// The same error should be produced with repr
#[repr(u8, u32)]
enum SeveralIntegerTypesRepr {
    Variant,
}

fn main() {}
