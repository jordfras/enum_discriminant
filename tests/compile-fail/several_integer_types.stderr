error[E0566]: conflicting representation hints
 --> tests/compile-fail/several_integer_types.rs:3:16
  |
3 | #[discriminant(u8, u32)]
  |                ^^  ^^^
  |
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #68585 <https://github.com/rust-lang/rust/issues/68585>
  = note: `#[deny(conflicting_repr_hints)]` on by default

error[E0566]: conflicting representation hints
 --> tests/compile-fail/several_integer_types.rs:9:8
  |
9 | #[repr(u8, u32)]
  |        ^^  ^^^
  |
  = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
  = note: for more information, see issue #68585 <https://github.com/rust-lang/rust/issues/68585>
